### ISSUE TITLE
请教有关sdpn的损失函数

### ISSUE THREAD

**Issue by evelynlee999** [alibaba-damo-academy/3D-Speaker#259]
你好，我想问一下有关sdpn的损失函数

看论文里写的L=L_ce + tau L_DR

但是代码中是
`loss = ploss + config.memax_weight * me_max + config.koleo_loss_weight * ke_loss`
其中config.koleo_loss_weight=0.1

你提出的L_DR在代码中用到的其实就是Koleo损失是吗？因为你的公式是：

<img width="458" height="80" alt="Image" src="https://github.com/user-attachments/assets/7838a369-5483-4d68-8965-7f6d1f059ba9" />

DINOv2中用到的是：

<img width="321" height="93" alt="Image" src="https://github.com/user-attachments/assets/5ca10aea-0824-4c65-b0be-8e798846ed9c" />

<img width="1276" height="83" alt="Image" src="https://github.com/user-attachments/assets/4d50a046-f966-4054-ac10-13ca333a7438" />

**Comment by evelynlee999** [comment]
还想请教一下，sdpn是直接训就行，还是要基于speech_sdpn_ecapa_tdnn_sv_en_voxceleb_16k这个模型？这个是你们保存的eer最好的那个是吗？

**Comment by yfchenlucky** [comment]
1. 对，L_DR在代码中是Koleo损失。
2. 直接使用训练脚本训练模型即可，不需要加载任何预训练模型。
3. 是文章中对应的模型。

**Comment by evelynlee999** [comment]
好的好的，谢谢哦～发自我的 iPhone在 2026年3月27日，18:09，Chen Yafeng ***@***.***> 写道：﻿yfchenlucky left a comment (modelscope/3D-Speaker#259)

对，L_DR在代码中是Koleo损失。
直接使用训练脚本训练模型即可，不需要加载任何预训练模型。
是文章中对应的模型。
